esp-hal/src/uart.rs

```diff
@@ -528,25 +531,14 @@ pub struct AtCmdConfig {
     pub char_num: Option<u8>,
```

Comment:
I wonder if we should make this `Option<core::num::NonZeroU8>` - `Some(0)` doesn't make sense and if `None` we use `1`

Would save us two bytes - not sure if it's worth it - makes things a bit less nice in user code

Reply:
This can also just be `u8`, can't it?

Reply:
yes - sounds good

Reply:
(probably needs some explanation in the doc-comment)

Reply:
Then we need to validate `AtCmdConfig` if `char_num` is not 0 which is related https://github.com/esp-rs/esp-hal/issues/2839 and probably should not be done in this PR (?)

Reply:
0 might be a perfectly reasonable value of "I don't want to detect AT commands"

Reply:
`0` just means the same as currently `None` - shouldn't change anything?

BUT I just realized that some chips (e.g. ESP32 at least) offer more than 16 bits for the counts and the time-out.
Big question before stabilization would be if we are fine with artificially limiting this or not

Reply:
oh and I think it solves a problem: currently we cannot undo the at-cmd-config once set? After the change the user could set `char_num` to `0` and effectivly disable it